openvino_xai/methods/base.py

```diff
@@ -23,6 +25,7 @@ def __init__(
         self._model_compiled = None
         self.preprocess_fn = preprocess_fn
         self._device_name = device_name
+        self.metadata: Dict[Task, Any] = collections.defaultdict(dict)
```

Comment:
Could you elaborate why the `Task` is the key?
Assuming a single method supports multiple tasks?
Dict[str, Any] might be more general metadata structure.

Reply:
> Assuming a single method supports multiple tasks?

Do we have such a case?

> Dict[str, Any] might be more general metadata structure.

What is a string in here?

metadata should follow some schema. So that visualizer and potentially some other explanation consumers can process it.

Schema differs for classification and detection tasks. That is why I implemented metadata to be Dict[Task, Dict[target, metadata_info]].
Therefore e.g. Visualizer just checks if there is metadata for detection, and if there is, it will process it.

Reply:
I was to ask whether you were assuming "a single method supporting multiple tasks" when I saw the task key. The only difference in input for XAI between CLS vs DET is the bbox info, then we may choose to add bbox as an optional parameter for explainer and/or visualizer.
BTW, if you intend to add some other metadata for CLS, it a different story. Let's discuss more after functional implementation.